server/data/visitSchedulerApiClient.ts

```diff
@@ -100,41 +100,42 @@ class VisitSchedulerApiClient {
     })
   }
 
-  updateVisit(visitData: VisitSessionData, visitStatus: string): Promise<Visit> {
-    const visitContact = visitData.mainContact
+  changeReservedVisit(visitSessionData: VisitSessionData): Promise<Visit> {
+    const visitContact = visitSessionData.mainContact
       ? {
-          telephone: visitData.mainContact.phoneNumber,
-          name: visitData.mainContact.contactName
-            ? visitData.mainContact.contactName
-            : visitData.mainContact.contact.name,
+          telephone: visitSessionData.mainContact.phoneNumber,
+          name: visitSessionData.mainContact.contactName
+            ? visitSessionData.mainContact.contactName
+            : visitSessionData.mainContact.contact.name,
         }
       : undefined
     const mainContactId =
-      visitData.mainContact && visitData.mainContact.contact ? visitData.mainContact.contact.personId : null
+      visitSessionData.mainContact && visitSessionData.mainContact.contact
+        ? visitSessionData.mainContact.contact.personId
+        : null
 
     return this.restclient.put({
-      path: `/visits/${visitData.visitReference}`,
-      data: <UpdateVisitRequestDto>{
-        prisonerId: visitData.prisoner.offenderNo,
-        prisonId: this.prisonId,
-        visitRoom: visitData.visit.visitRoomName,
-        visitType: this.visitType,
-        visitStatus,
-        visitRestriction: visitData.visitRestriction,
-        startTimestamp: visitData.visit.startTimestamp,
-        endTimestamp: visitData.visit.endTimestamp,
+      path: `/visits/${visitSessionData.applicationReference}/slot/change`,
+      data: <ChangeReservedVisitSlotRequestDto>{
+        visitRestriction: visitSessionData.visitRestriction,
+        startTimestamp: visitSessionData.visit.startTimestamp,
+        endTimestamp: visitSessionData.visit.endTimestamp,
         visitContact,
-        visitors: visitData.visitors.map(visitor => {
+        visitors: visitSessionData.visitors.map(visitor => {
           return {
             nomisPersonId: visitor.personId,
             visitContact: visitor.personId === mainContactId,
           }
         }),
-        visitorSupport: visitData.visitorSupport,
+        visitorSupport: visitSessionData.visitorSupport,
       },
     })
   }
 
+  bookVisit(visitSessionData: VisitSessionData): Promise<Visit> {
```

Comment:
Is there any point passing all the data to this method when it only uses the reference?

Reply:
Good point 🤦‍♂️ ! I'll tweak it...

Reply:
OK, changed to just work with `application` reference - which is all it needs.